### DISCUSSION TITLE
Boinccmd Fails After 8.0.x Install on Fedora 39

### DISCUSSION THREAD

**Discussion by gj82854** [BOINC/boinc#5622] [General]
After upgrading Boinc to 8.0.x I noticed that all my boinccmd crontab entries were failing (after fixing the change from /usr/bin to /usr/local/bin). Logs contained:

boinccmd --project gene.disi.unitn.it/test/ update
Can't get RPC password: gui_rpc_auth.cfg not found.  See https://boinc.berkeley.edu/gui_rpc.php
Only operations not requiring authorization will be allowed.
Operation failed: authentication error

There is an existing gui_rpc_auth.cfg file in /var/lib/boinc. In the old version 7 world of boinc there was a file in the /etc/boinc-client directory (config.properties) that pointed to the boinc DATA_DIR. That file no longer exists in the version 8 world. I tried adding the gui_rpc_auth.cfg to the /etc/boinc-client directory but that failed. Tried running under the root account just in case it needed access to the systemd environment to find the auth file but that also failed. Also tried exporting DATA_DIR as an environment variable but that failed too. The only way I was able to get the boinccmd to run was to execute the command with /var/lib/boinc as the current working directory. It seems boinccmd can't find the gui_rpc_auth.cfg file after the upgrade to version 8 of boinc. 

I opened a discussion item first to see if anyone else is having the same problem. This is happening on all 5 of my Fedora environments. I tested on Ubuntu 22.04 LTS and it works fine but there are files in the /etc/boinc-client directory under Ubuntu. The /etc/boinc-client directory is empty on Fedora. How does boinccmd find the gui_rpc_auth.cfg file on Fedora systems?

**Comment by AenBleidd** [comment]
Thank you for the report.
I'll fix that in scope with https://github.com/BOINC/boinc/issues/5564

**Comment by gj82854** [comment]
I have tested this in the 8.0.2 release and the problem still exists. In Fedora there is a file called /etc/boinc-client that used to contain a variable that pointed to the BOINC data directory. That file is now empty (except for some comments). In Ubuntu, the /etc directory contains the gui_rpc_auth.cfg file and the /var/lib/boinc-client contains a link to the file. everything seems to work in Ubuntu but not Fedora. In there something in the systemd configuration on Ubuntu that allows programs to find the data directory but is absent in Fedora? Additionally, it would seem to be wise to try and get the Linux environments to be consistent such as:

Either use /var/lib/boinc (Fedora) or /var/lib/boinc-client (Ubuntu)
Either use files in /etc (Ubuntu) or don't use files in /etc (Fedora)
